src/FTPWindow.cpp

```diff
@@ -379,8 +378,7 @@ void TFTPWindow::FtpReportMsgIncoming(BMessage *msg)
 			if (fFtpLooper) {
 				fFtpLooper->Abort();
 				BMessenger(fFtpLooper).SendMessage(B_QUIT_REQUESTED);
-				//->Lock();
-//				fFtpLooper->Quit();
+				delete fFtpLooper;
```

Comment:
If I understand correctly, B_QUIT_REQUESTED makes the looper call Quit, and that deletes it, so this line is redundant.

Reply:
Yes, the looper will self-destroy itself.

Reply:
this is still pending.

Reply:
Actually, when I worked on this commit I just mindlessly removed code that was commented out and I saw that there's a memory leak over there, so I wrote "delete fFtpLooper;".

But now I started the question myself. As far as I know this BMessenger thing works asynchronously. Thus if I send a message it won't block the code execution until the message gets handled so it would cause problems if the object was deleted. Or am I wrong?

Reply:
B_QUIT_REQUESTED is supposed to have the fFtpLooper thread execute "delete this", so there is no leak.